triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRunnerImpl.scala

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.engine.trigger
+
+import akka.actor.typed.{Behavior, PostStop}
+import akka.actor.typed.PostStop
+import akka.actor.typed.PreRestart
+import akka.actor.typed.scaladsl.Behaviors
+import akka.stream.{KillSwitch, KillSwitches, Materializer}
+import com.daml.ledger.api.refinements.ApiTypes.Party
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+import scalaz.syntax.tag._
+import com.daml.lf.CompiledPackages
+import com.daml.grpc.adapter.{ExecutionSequencerFactory}
+import com.daml.ledger.api.refinements.ApiTypes.ApplicationId
+import com.daml.ledger.api.v1.event.CreatedEvent
+import com.daml.ledger.api.v1.ledger_offset.LedgerOffset
+import com.daml.ledger.client.LedgerClient
+import com.daml.ledger.client.configuration.{
+  CommandClientConfiguration,
+  LedgerClientConfiguration,
+  LedgerIdRequirement
+}
+
+object TriggerRunnerImpl {
```

Comment:
Maybe this is an akka-pattern I’m not aware off but I’ve only seen `Impl` be used if you have an abstract interface that you are implementing. This is not what is going on here so it seems rather confusing.

Reply:
Maybe the nomenclature is non-idiomatic for Scala (it's common in C++ e.g. `pImpl`) but at the end of the day, it's nothing more than the classic "handle-body" or "letter-envelope" pattern.